### ISSUE TITLE
[BUG]. DefaultForecaster returns naive forecast. 

### ISSUE THREAD

**Issue by microprediction** [salesforce/Merlion#25]
**Describe the bug**
DefaultForecaster returns naive estimate when one might expect otherwise. 

**To Reproduce**
See [notebook](https://github.com/microprediction/timeseries-notebooks/blob/main/merlion_test_flat_line.ipynb)

**Expected behavior**
It may be fine. Perhaps it awaits more data or there is a bug in the usage. I'm yet to trace in and the example was created by a fellow contributor. 

**Desktop (please complete the following information):**
Colab.

**Comment by chenghaoliu89** [comment]
Hi @microprediction, thanks for sharing this case. The default forecaster (ETS) will detect seasonality automatically. Here, we make a constraint that the maximum periodicity should be smaller than `max(min(int(10 * np.log10(x.shape[0])), x.shape[0] - 1), 40)`. The reason is we should guarantee sufficient repetition of seasonal pattern. As a large periodicity indicates a large model complexity, which requires more data to train, and it is easy to over-fitting.   

From the given data, I found  the statistically significant periodicity is 513, so it will assign none seasonal pattern and output naive forecast. If we set the periodicity to be  513 manually, the prediction results is as follows, we can see the result is also not good due to insufficient data, and it is difficult to say which one is better in terms of RMSE.  

![image](https://user-images.githubusercontent.com/74166079/137155963-53b3f05f-0fc1-40f0-8063-07df80f482b6.png)
